### PR TITLE
Feat/random channel dropout

### PR DESCRIPTION
**Pull request by bohdanhlovatskyi**
#### Changes

Fixes # (issue)

#### Type of change

- [+] 📚  Documentation Update
- [+] 🧪 Tests Cases
- [+] 🔬 New feature (non-breaking change which adds functionality)

#### Checklist
- [+] My code follows the style guidelines of this project
- [+] I have performed a self-review of my own code
- [+] I have commented my code, particularly in hard-to-understand areas
- [+] I have made corresponding changes to the documentation
- [+] My changes generate no new warnings

### CODE REVIEW

File: kornia/augmentation/_2d/intensity/channel_dropout.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    def generate_parameters(self, shape: torch.Size) -> Dict[str, Tensor]:
+        B, C, _, _ = shape
+        if C <= 1:
+            raise ValueError("Channel dropout can be applied only to multichannel images")
+
+        if self.max_channels >= C:
+            raise ValueError("Cannot drop all channels in ChannelDropout.")
+
+        # [B] - for each batch drop random amount of channels
+        num_drop_channels = torch.randint(low=self.min_channels, high=self.max_channels + 1, size=(B,))
```
Comment (edgarriba) [inline kornia/augmentation/_2d/intensity/channel_dropout.py:61]: i think here the preferred way is to use the Samplers API
https://github.com/kornia/kornia/blob/master/kornia/augmentation/_2d/intensity/random_snow.py#L48

/cc @shijianjian

File: test/augmentation/test_augmentation.py [hunk 2]
```diff
@@ -3483,6 +3485,50 @@ def test_module(self, device, dtype):
...
+        aug1 = RandomChannelDropout(channel_drop_range=(1, 2), p=1.0)
+        out1 = aug1(img1)
+
+        aug2 = RandomChannelDropout(channel_drop_range=(1, 1), p=1.0)
+        out2 = aug2(img2)
+
+        self.assert_close(out1, out_expected1)
+        self.assert_close(out2, out_expected2)
+
+    @pytest.mark.skip(reason="not implemented yet")
```
Comment (edgarriba) [inline test/augmentation/test_augmentation.py:3511]: i you don't implement maybe it's not worth to subclass from BaseTester. @johnnv1 we might rethink about different test cases

File: kornia/augmentation/_2d/intensity/channel_dropout.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    def apply_transform(
+        self, input: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any], transform: Optional[Tensor] = None
+    ) -> Tensor:
+        KORNIA_CHECK_IS_TENSOR(input)
+        KORNIA_CHECK_SHAPE(input, ['B', 'C', 'H', 'W'])
+
+        _, C, _, _ = input.shape
+        if C <= 1:
+            raise ValueError("Channel dropout can be applied only to multichannel images")
```
Comment (edgarriba) [inline kornia/augmentation/_2d/intensity/channel_dropout.py:84]: KORNIA_CHECK ?

File: kornia/augmentation/_2d/intensity/channel_dropout.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        )
+
+        channels_to_drop_one_hot = channels_to_drop_one_hot.gather(dim=1, index=channels_to_drop)
+
+        return dict(channels_to_drop=channels_to_drop_one_hot)
+
+    def apply_transform(
+        self, input: Tensor, params: Dict[str, Tensor], flags: Dict[str, Any], transform: Optional[Tensor] = None
+    ) -> Tensor:
+        KORNIA_CHECK_IS_TENSOR(input)
```
Comment (edgarriba) [inline kornia/augmentation/_2d/intensity/channel_dropout.py:79]: is tensor is included in check_shape i think

File: kornia/augmentation/_2d/intensity/channel_dropout.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            raise ValueError("Channel dropout can be applied only to multichannel images")
+
+        if self.max_channels >= C:
+            raise ValueError("Cannot drop all channels in ChannelDropout.")
+
+        # [B] - for each batch drop random amount of channels
+        num_drop_channels = torch.randint(low=self.min_channels, high=self.max_channels + 1, size=(B,))
+
+        # B, C
+        channels_to_drop = torch.argsort(torch.rand(B, C))
```
Comment (edgarriba) [inline kornia/augmentation/_2d/intensity/channel_dropout.py:64]: same for the samplers

File: kornia/augmentation/_2d/intensity/channel_dropout.py (hunk 1 above)
Comment (bohdanhlovatskyi) [reply kornia/augmentation/_2d/intensity/channel_dropout.py]: +

File: test/augmentation/test_augmentation.py (hunk 2 above)
Comment (bohdanhlovatskyi) [reply test/augmentation/test_augmentation.py:3511]: assert_close does not work without BaseTester, so, for consistency, I assume it is ok (?)

File: kornia/augmentation/_2d/intensity/channel_dropout.py (hunk 5 above)
Comment (bohdanhlovatskyi) [reply kornia/augmentation/_2d/intensity/channel_dropout.py:73]: Not sure whether it is possible to rewrite this torch.argsort(torch.rand(B, C)) nicely via Sample API, ChannelShuffle does this in the same way

Comment (shijianjian) [reply kornia/augmentation/_2d/intensity/channel_dropout.py:73]: I feel like you may have a custom `RandomGenerator`. For example,

https://github.com/kornia/kornia/blob/master/kornia/augmentation/random_generator/_2d/gaussian_blur.py

**Pull request closed without merging**